### PR TITLE
mosh: add mosh and protobuf

### PR DESCRIPTION
**Pull request by jslick**
Please fill out the template.

Read: https://kiss.armaanb.net/wiki/kiss/style-guide

---

## Description of package

* mosh is a remote shell utilizing UDP instead of TCP featuring performance enhancements and automatic reconnects
* mosh depends on protobuf

## Installed manifest of package

```
/var/db/kiss/installed/protobuf/version
/var/db/kiss/installed/protobuf/sources
/var/db/kiss/installed/protobuf/manifest
/var/db/kiss/installed/protobuf/depends
/var/db/kiss/installed/protobuf/checksums
/var/db/kiss/installed/protobuf/build
/var/db/kiss/installed/protobuf/
/var/db/kiss/installed/
/var/db/kiss/
/var/db/
/var/
/usr/share/vim/vimfiles/syntax/proto.vim
/usr/share/vim/vimfiles/syntax/
/usr/share/vim/vimfiles/
/usr/share/vim/
/usr/share/
/usr/lib/pkgconfig/protobuf.pc
/usr/lib/pkgconfig/protobuf-lite.pc
/usr/lib/pkgconfig/
/usr/lib/libprotoc.so.25.0.0
/usr/lib/libprotoc.so.25
/usr/lib/libprotoc.so
/usr/lib/libprotoc.a
/usr/lib/libprotobuf.so.25.0.0
/usr/lib/libprotobuf.so.25
/usr/lib/libprotobuf.so
/usr/lib/libprotobuf.a
/usr/lib/libprotobuf-lite.so.25.0.0
/usr/lib/libprotobuf-lite.so.25
/usr/lib/libprotobuf-lite.so
/usr/lib/libprotobuf-lite.a
/usr/lib/
/usr/include/google/protobuf/wrappers.proto
/usr/include/google/protobuf/wrappers.pb.h
/usr/include/google/protobuf/wire_format_lite.h
/usr/include/google/protobuf/wire_format.h
/usr/include/google/protobuf/util/type_resolver_util.h
/usr/include/google/protobuf/util/type_resolver.h
/usr/include/google/protobuf/util/time_util.h
/usr/include/google/protobuf/util/message_differencer.h
/usr/include/google/protobuf/util/json_util.h
/usr/include/google/protobuf/util/field_mask_util.h
/usr/include/google/protobuf/util/field_comparator.h
/usr/include/google/protobuf/util/delimited_message_util.h
/usr/include/google/protobuf/util/
/usr/include/google/protobuf/unknown_field_set.h
/usr/include/google/protobuf/type.proto
/usr/include/google/protobuf/type.pb.h
/usr/include/google/protobuf/timestamp.proto
/usr/include/google/protobuf/timestamp.pb.h
/usr/include/google/protobuf/text_format.h
/usr/include/google/protobuf/stubs/template_util.h
/usr/include/google/protobuf/stubs/strutil.h
/usr/include/google/protobuf/stubs/stringpiece.h
/usr/include/google/protobuf/stubs/stl_util.h
/usr/include/google/protobuf/stubs/status.h
/usr/include/google/protobuf/stubs/port.h
/usr/include/google/protobuf/stubs/platform_macros.h
/usr/include/google/protobuf/stubs/once.h
/usr/include/google/protobuf/stubs/mutex.h
/usr/include/google/protobuf/stubs/map_util.h
/usr/include/google/protobuf/stubs/macros.h
/usr/include/google/protobuf/stubs/logging.h
/usr/include/google/protobuf/stubs/hash.h
/usr/include/google/protobuf/stubs/common.h
/usr/include/google/protobuf/stubs/casts.h
/usr/include/google/protobuf/stubs/callback.h
/usr/include/google/protobuf/stubs/bytestream.h
/usr/include/google/protobuf/stubs/
/usr/include/google/protobuf/struct.proto
/usr/include/google/protobuf/struct.pb.h
/usr/include/google/protobuf/source_context.proto
/usr/include/google/protobuf/source_context.pb.h
/usr/include/google/protobuf/service.h
/usr/include/google/protobuf/repeated_field.h
/usr/include/google/protobuf/reflection_ops.h
/usr/include/google/protobuf/reflection.h
/usr/include/google/protobuf/port_undef.inc
/usr/include/google/protobuf/port_def.inc
/usr/include/google/protobuf/port.h
/usr/include/google/protobuf/parse_context.h
/usr/include/google/protobuf/metadata_lite.h
/usr/include/google/protobuf/metadata.h
/usr/include/google/protobuf/message_lite.h
/usr/include/google/protobuf/message.h
/usr/include/google/protobuf/map_type_handler.h
/usr/include/google/protobuf/map_field_lite.h
/usr/include/google/protobuf/map_field_inl.h
/usr/include/google/protobuf/map_field.h
/usr/include/google/protobuf/map_entry_lite.h
/usr/include/google/protobuf/map_entry.h
/usr/include/google/protobuf/map.h
/usr/include/google/protobuf/io/zero_copy_stream_impl_lite.h
/usr/include/google/protobuf/io/zero_copy_stream_impl.h
/usr/include/google/protobuf/io/zero_copy_stream.h
/usr/include/google/protobuf/io/tokenizer.h
/usr/include/google/protobuf/io/strtod.h
/usr/include/google/protobuf/io/printer.h
/usr/include/google/protobuf/io/io_win32.h
/usr/include/google/protobuf/io/gzip_stream.h
/usr/include/google/protobuf/io/coded_stream.h
/usr/include/google/protobuf/io/
/usr/include/google/protobuf/implicit_weak_message.h
/usr/include/google/protobuf/has_bits.h
/usr/include/google/protobuf/generated_message_util.h
/usr/include/google/protobuf/generated_message_table_driven.h
/usr/include/google/protobuf/generated_message_reflection.h
/usr/include/google/protobuf/generated_enum_util.h
/usr/include/google/protobuf/generated_enum_reflection.h
/usr/include/google/protobuf/field_mask.proto
/usr/include/google/protobuf/field_mask.pb.h
/usr/include/google/protobuf/extension_set_inl.h
/usr/include/google/protobuf/extension_set.h
/usr/include/google/protobuf/empty.proto
/usr/include/google/protobuf/empty.pb.h
/usr/include/google/protobuf/dynamic_message.h
/usr/include/google/protobuf/duration.proto
/usr/include/google/protobuf/duration.pb.h
/usr/include/google/protobuf/descriptor_database.h
/usr/include/google/protobuf/descriptor.proto
/usr/include/google/protobuf/descriptor.pb.h
/usr/include/google/protobuf/descriptor.h
/usr/include/google/protobuf/compiler/ruby/ruby_generator.h
/usr/include/google/protobuf/compiler/ruby/
/usr/include/google/protobuf/compiler/python/python_generator.h
/usr/include/google/protobuf/compiler/python/
/usr/include/google/protobuf/compiler/plugin.proto
/usr/include/google/protobuf/compiler/plugin.pb.h
/usr/include/google/protobuf/compiler/plugin.h
/usr/include/google/protobuf/compiler/php/php_generator.h
/usr/include/google/protobuf/compiler/php/
/usr/include/google/protobuf/compiler/parser.h
/usr/include/google/protobuf/compiler/objectivec/objectivec_helpers.h
/usr/include/google/protobuf/compiler/objectivec/objectivec_generator.h
/usr/include/google/protobuf/compiler/objectivec/
/usr/include/google/protobuf/compiler/js/well_known_types_embed.h
/usr/include/google/protobuf/compiler/js/js_generator.h
/usr/include/google/protobuf/compiler/js/
/usr/include/google/protobuf/compiler/java/java_names.h
/usr/include/google/protobuf/compiler/java/java_generator.h
/usr/include/google/protobuf/compiler/java/
/usr/include/google/protobuf/compiler/importer.h
/usr/include/google/protobuf/compiler/csharp/csharp_names.h
/usr/include/google/protobuf/compiler/csharp/csharp_generator.h
/usr/include/google/protobuf/compiler/csharp/
/usr/include/google/protobuf/compiler/cpp/cpp_generator.h
/usr/include/google/protobuf/compiler/cpp/
/usr/include/google/protobuf/compiler/command_line_interface.h
/usr/include/google/protobuf/compiler/code_generator.h
/usr/include/google/protobuf/compiler/
/usr/include/google/protobuf/arenastring.h
/usr/include/google/protobuf/arena_impl.h
/usr/include/google/protobuf/arena.h
/usr/include/google/protobuf/api.proto
/usr/include/google/protobuf/api.pb.h
/usr/include/google/protobuf/any.proto
/usr/include/google/protobuf/any.pb.h
/usr/include/google/protobuf/any.h
/usr/include/google/protobuf/
/usr/include/google/
/usr/include/
/usr/bin/protoc
/usr/bin/
/usr/
```

---

```
/var/db/kiss/installed/mosh/version
/var/db/kiss/installed/mosh/sources
/var/db/kiss/installed/mosh/manifest
/var/db/kiss/installed/mosh/depends
/var/db/kiss/installed/mosh/checksums
/var/db/kiss/installed/mosh/build
/var/db/kiss/installed/mosh/
/var/db/kiss/installed/
/var/db/kiss/
/var/db/
/var/
/usr/share/man/man1/mosh.1
/usr/share/man/man1/mosh-server.1
/usr/share/man/man1/mosh-client.1
/usr/share/man/man1/
/usr/share/man/
/usr/share/bash-completion/completions/mosh
/usr/share/bash-completion/completions/
/usr/share/bash-completion/
/usr/share/
/usr/bin/mosh-server
/usr/bin/mosh-client
/usr/bin/mosh
/usr/bin/
/usr/
```

## New package

- [X] Latest upstream version.
- [X] I agree to maintain this package.
- [X] I agree to notify the @kiss-community maintainers if I can no longer maintain this package.

## Existing package

N/A

### CODE REVIEW

**Comment by noocsharp** [comment]
Could you split this into two commits, one for mosh and one for protobuf?

**Comment by jslick** [comment]
> Could you split this into two commits, one for mosh and one for protobuf?

yep, done!

File: community/mosh/depends [hunk 1]
```diff
@@ -0,0 +1,3 @@
+openssh
+perl
```
Comment (git-bruh) [inline community/mosh/depends:2]: Does the package need perl or is it just a build time dependency ?

File: community/protobuf/build [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#!/bin/sh -e
+
+export DESTDIR="$1"
+
+./configure --prefix=/usr
+make
+
+make DESTDIR="$DESTDIR" install
+install -vDm 644 LICENSE            -t "$1/usr/share/licenses/protobuf"
+install -vDm 644 editors/proto.vim  -t "$1/usr/share/vim/vimfiles/syntax"
```
Comment (git-bruh) [inline community/protobuf/build:10]: The verbose (`-v`) flag is usually not used in build files

**Review by git-bruh — CHANGES_REQUESTED** [review]

File: community/mosh/depends (hunk 1 above)
Comment (jslick) [reply community/mosh/depends:2]: yes, /usr/bin/mosh is written in perl, so it is needed for mosh to function

File: community/protobuf/build (hunk 2 above)
Comment (jslick) [reply community/protobuf/build]: updated

**Comment by git-bruh** [comment]
@jslick according to the style guide, `/usr`, `/etc` and the `conf/bash-completion/completions/mosh` directory should not be quoted as they have no variables in them, could you send a PR with these changes ?

**Pull request merged**